ci-operator/templates/build-azure-image.yaml

```diff
@@ -0,0 +1,315 @@
+
+        # added timeout as this seems to loop infinitely awaiting cluster deployment
+        timeout 30m ./hack/create.sh ${RESOURCE_GROUP}-e2e
+
+        # if completed pass generated data to other stage
+        cp -r /go/src/github.com/openshift/openshift-azure/_data /tmp/shared/
+
+        touch /tmp/shared/e2e_tests
+
+    - name: tagimage
+      image: ${IMAGE_ANSIBLE}
+      resources:
+        requests:
+          cpu: 1
+          memory: 300Mi
+        limits:
+          cpu: 3
+          memory: 4Gi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /usr/share/ansible/openshift-ansible/inventory/dynamic/injected
+      env:
+      - name: HOME
+        value: /usr/share/ansible/openshift-ansible
+      - name: TYPE
+        value: azure
+      command:
+      - /usr/local/bin/entrypoint-provider
+      args:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        # error handling and sync code
+        set -euo pipefail
+
+        trap '[ "$?" == 0 ] || touch /tmp/shared/exit' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        # wait until testing our image has succeeded
+        while true; do
+          if [[ ! -f  /tmp/shared/e2e_tests ]]; then
+            sleep 15 & wait
+            continue
+          fi
+          # if we got admin kubeconfig file with failure marker, ignore. Teardown is in progress.
+          if [[ -f  /tmp/shared/exit ]]; then
+             exit 1
+          fi
+          break
+        done
+        echo "Found image create marker file."
+
+        # Get resource group and image name
+        source /tmp/shared/environment
+
+        # Source azure creds
+        source /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/credentials
+
+        ansible-playbook \
+          -e openshift_azure_resource_group_name=$RESOURCE_GROUP \
+          -e openshift_azure_input_image_ns=images \
+          -e openshift_azure_input_image_name=$IMAGE_NAME \
+          playbooks/azure/openshift-cluster/tag_image_as_valid.yml
+
+        touch /tmp/shared/exit
+
+    # Performs cleanup of all created resources
+    - name: teardown
+      image: ${LOCAL_IMAGE_BIN}
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /etc/azure/credentials
+      env:
+      - name: TYPE
+        value: ${CLUSTER_TYPE}
+      - name: HOME
+        value: /tmp/shared/home
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        function teardown() {
+          set +e
+          mkdir -p "${HOME}"
+          export HOME=/tmp/shared
+          export DNS_DOMAIN=osadev.cloud
+          export DNS_RESOURCEGROUP=dns
+
+          cp -r /tmp/shared/_data /go/src/github.com/openshift/openshift-azure/
+          cd /go/src/github.com/openshift/openshift-azure/
+          source /etc/azure/credentials/secret
```

Comment:
You also need to source `/tmp/shared/environment` to get the resource group, right?